### PR TITLE
Products filters

### PR DESCRIPTION
**Pull request by Jay-Gadhiya**
Live link : https://silkshopy.vercel.app/

Add :  filter functionality on product listing page usign useContext and useReducer.
folder added : context, reducer.

### CODE REVIEW

File: silkshopy-ecommerce/src/pages/ProductListing/functions/sort-price.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const sortByPrice = (sortBy, priceRangeData) => {
+
+    if (sortBy === "LOW_TO_HIGH")
+        return priceRangeData.sort((a, b) => Number(a.price) - Number(b.price));
+
+    if (sortBy === "HIGH_TO_LOW")
+        return priceRangeData.sort((a, b) => Number(b.price) - Number(a.price));
```
Comment (RahullRawat) [inline silkshopy-ecommerce/src/pages/ProductListing/functions/sort-price.jsx]: Here you are using sort function and sort function mutates the original array. So my suggestion is that you can use spread operator while sorting.

File: silkshopy-ecommerce/src/pages/ProductListing/functions/fetch-data.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const [product, setProduct] = useState([]);
+
+  useEffect(
+    () =>
+      (async () => {
+        try {
+          const res = await axios.get("/api/products");
+          setProduct(() => res.data.products);
+        } catch (error) {
+          console.log(error);
```
Comment (RahullRawat) [inline silkshopy-ecommerce/src/pages/ProductListing/functions/fetch-data.jsx]: Remove this unwanted console.log

File: silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx [hunk 3]
```diff
@@ -19,19 +27,39 @@ const Filters = () => {
...
+                />
                 <label htmlFor="low-to-high">Price : Low to High</label>
             </div>
 
             <div className="sort-price flex">
-                <input type="radio" name="price" id="low-to-high" />
+                <input 
+                type="radio" 
+                name="price" 
+                id="low-to-high"
```
Comment (RahullRawat) [inline silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx]: Here your id is different from labelFor attribute. Please correct it

File: silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx [hunk 4]
```diff
@@ -64,22 +116,46 @@ const Filters = () => {
             <label htmlFor="Sort by" className="font-style sort-by-label">Rating</label>
 
             <div className="sort-price flex">
-                <input type="radio" name="rating" id="low-to-high" />
+                <input 
+                type="radio" 
+                name="rating" 
+                id="low-to-high"
```
Comment (RahullRawat) [inline silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx]: Id is different

File: silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx [hunk 5]
```diff
@@ -64,22 +116,46 @@ const Filters = () => {
...
+                />
                 <label htmlFor="4 Star & above">4 Star & above</label>
             </div>
 
             <div className="sort-price flex">
-                <input type="radio" name="rating" id="low-to-high" />
+                <input 
+                type="radio" 
+                name="rating" 
+                id="low-to-high"
```
Comment (RahullRawat) [inline silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx]: Here

File: silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx [hunk 6]
```diff
@@ -64,22 +116,46 @@ const Filters = () => {
...
+                 />
                 <label htmlFor="3 Star & above">3 Star & above</label>
             </div>
 
             <div className="sort-price flex">
-                <input type="radio" name="rating" id="low-to-high" />
+                <input 
+                type="radio" 
+                name="rating" 
+                id="low-to-high"
```
Comment (RahullRawat) [inline silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx]: Same here

File: silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx [hunk 7]
```diff
@@ -64,22 +116,46 @@ const Filters = () => {
...
+                />
                 <label htmlFor="2 Star & above">2 Star & above</label>
             </div>
 
             <div className="sort-price flex">
-                <input type="radio" name="rating" id="low-to-high" />
+                <input 
+                type="radio" 
+                name="rating" 
+                id="low-to-high"
```
Comment (RahullRawat) [inline silkshopy-ecommerce/src/pages/ProductListing/components/Filter.jsx]: Same here

**Review by RahullRawat — APPROVED** [review]
Apart from suggestions code looks good

**Review by samyakshah3008 — APPROVED** [review]
Loved you clean UI specailly product page 
Proper folder structure and semantic html tags ✅ 
Description you provided for preview helped for better review 

Good for approval ✅

**Pull request merged**